private-media-framework/draft-ietf-perc-private-media-framework.md

```diff
@@ -711,3 +698,362 @@ a co-author.
+| HBH Key (MD X<=>MD Y)|    No      |  Yes  |  Yes  |     No     |
++----------------------+------------+---------------+------------+
+~~~
+Figure: Keys per Entity
+
+# PERC Packet Format {#packetformat}
+
+This appendix is intended to aid the reader in understanding the PERC media
+packet format by walking through the steps required to construct the packet
+at each step.
+
+## PERC Packet Formation Steps
+
+### Step 1: Formulate an RTP Packet
+
+First, we start with a plaintext RTP packet that is formed by the
+transmitting endpoint, but has not undergone any kind of cryptographic
+transform.  It would look like the one shown in [@fig-plaintext-packet].
+
+{#fig-plaintext-packet align="center"}
+~~~
++------------------------------+
+|          RTP Header          |
++------------------------------+
+|                              |
+|       Plaintext Payload      |
+|                              |
++------------------------------+
+~~~
+Figure: Plaintext RTP Packet before Encryption
+
+While not explicitly called out, the "RTP Header" in the diagrams in this
+appendix would include any allowable or required RTP header extensions.
+
+### Step 2: Apply the E2E Cryptographic Transform
+
+The endpoint would then take this packet and pass it to the code responsible
+for applying the SRTP cryptographic transform for PERC.  In Double
+[@!I-D.ietf-perc-double], this is looked at by the endpoint as a single
+operation, but since two different keys are utilized and since some readers
+will be responsible for writing that software, the steps are presented
+separately.
+
+The first step in applying the Double transform is to encrypt the packet
+using the endpoint's E2E SRTP master key.  This is the "inner" transform.
+Once that cryptographic transform has been applied using the E2E key, the
+packet would look like the one shown in [@fig-e2e-protected-packet].
+
+{#fig-e2e-protected-packet align="center"}
+~~~
++------------------------------+
+|          RTP Header          |
++------------------------------+
+|                              |
+|      Ciphertext Payload      |
+|                              |
++------------------------------+
+|      Authentication Tag      |
++------------------------------+
+~~~
+Figure: Packet after Cryptographic Transform using E2E Key
+
+Note that the E2E key is not used directly.  Rather, SRTP ([@RFC3711])
+specifies in Section 4.3.1 how to derive the session encryption key
+(k_e) from the SRTP master key.  It is k_e that is used to actually
+encrypt the packet.  This is true both this step and the next step where
+the HBH key is used.  To keep the text simple here, the reader should
+understand that E2E keys or HBH keys must be passed through the key
+derivation function specified in RFC 3711 or elsewhere depending on the
+cipher.
+
+With certain cryptographic transforms like AES-GCM ([@RFC7714]), use of
+a separate authentication tag is discouraged (as it is redundant).
```

Comment:
I'm finding this confusing on the topic of how the auth tag is handled for the current aes-gcm in double.